packages/dataviews/src/field-types/integer.tsx

```diff
@@ -1,89 +1,13 @@
-/**
- * WordPress dependencies
- */
-import { __ } from '@wordpress/i18n';
-
 /**
  * Internal dependencies
  */
-import type {
-	DataViewRenderFieldProps,
-	SortDirection,
-	NormalizedField,
-	FieldTypeDefinition,
-} from '../types';
-import { renderFromElements } from '../utils';
-import {
-	OPERATOR_IS,
-	OPERATOR_IS_NOT,
-	OPERATOR_LESS_THAN,
-	OPERATOR_GREATER_THAN,
-	OPERATOR_LESS_THAN_OR_EQUAL,
-	OPERATOR_GREATER_THAN_OR_EQUAL,
-	OPERATOR_IS_ANY,
-	OPERATOR_IS_NONE,
-	OPERATOR_IS_ALL,
-	OPERATOR_IS_NOT_ALL,
-	OPERATOR_BETWEEN,
-} from '../constants';
-
-function sort( a: any, b: any, direction: SortDirection ) {
-	return direction === 'asc' ? a - b : b - a;
-}
+import number, { getValidation } from './number';
+import type { FieldTypeDefinition } from '../types';
 
 export default {
-	sort,
-	isValid: {
-		custom: ( item: any, field: NormalizedField< any > ) => {
-			const value = field.getValue( { item } );
-			if (
-				! [ undefined, '', null ].includes( value ) &&
-				! Number.isInteger( value )
-			) {
-				return __( 'Value must be an integer.' );
-			}
-
-			if ( field?.elements ) {
-				const validValues = field.elements.map( ( f ) => f.value );
-				if ( ! validValues.includes( Number( value ) ) ) {
-					return __( 'Value must be one of the elements.' );
-				}
-			}
-
-			return null;
-		},
-	},
+	...number,
 	Edit: 'integer',
-	render: ( { item, field }: DataViewRenderFieldProps< any > ) => {
-		return field.elements
-			? renderFromElements( { item, field } )
-			: field.getValue( { item } );
-	},
-	enableSorting: true,
-	filterBy: {
-		defaultOperators: [
-			OPERATOR_IS,
-			OPERATOR_IS_NOT,
-			OPERATOR_LESS_THAN,
-			OPERATOR_GREATER_THAN,
-			OPERATOR_LESS_THAN_OR_EQUAL,
-			OPERATOR_GREATER_THAN_OR_EQUAL,
-			OPERATOR_BETWEEN,
-		],
-		validOperators: [
-			// Single-selection
-			OPERATOR_IS,
-			OPERATOR_IS_NOT,
-			OPERATOR_LESS_THAN,
-			OPERATOR_GREATER_THAN,
-			OPERATOR_LESS_THAN_OR_EQUAL,
-			OPERATOR_GREATER_THAN_OR_EQUAL,
-			OPERATOR_BETWEEN,
-			// Multiple-selection
-			OPERATOR_IS_ANY,
-			OPERATOR_IS_NONE,
-			OPERATOR_IS_ALL,
-			OPERATOR_IS_NOT_ALL,
-		],
+	isValid: {
```

Comment:
`isValid`, `Edit`, and `render` need to be different, and so I'd rather declare these types as separate components that don't know about each other. I appreciate the drive for reusability, but it is impacting readability too much for no gain in this case.